File: planet_wars/starter_packages/java/MyBot.java
import java.util.*;

public class MyBot {
    public static void DoTurn(PlanetWars pw, Random r) {
	for (int i = 0; i < pw.NumFleets(); ++i) {
	    if (pw.GetFleet(i).Owner() == 1) {
		System.out.println("go");
		return;
	    }
	}
	List<Integer> myPlanets = new ArrayList<Integer>();
	List<Integer> opPlanets = new ArrayList<Integer>();
	for (int i = 0; i < pw.NumPlanets(); ++i) {
	    Planet p = pw.GetPlanet(i);
	    if (p.Owner() == 1) {
		myPlanets.add(i);
	    } else {
		opPlanets.add(i);
	    }
	}
	if (myPlanets.size() > 0 && opPlanets.size() > 0) {
	    int source = myPlanets.get(r.nextInt(myPlanets.size()));
	    int destination = opPlanets.get(r.nextInt(opPlanets.size()));
	    int numShips = r.nextInt(pw.GetPlanet(source).NumShips());
	    System.out.println("" + source + " " +
			       destination + " " + numShips);
	}
	System.out.println("go");
    }

    public static void main(String[] args) {
	String line = "";
	String message = "";
	int c;
	try {
	    Random r = new Random();
	    while ((c = System.in.read()) >= 0) {
		switch (c) {
		case '\n':
		    if (line.equals("go")) {
			PlanetWars pw = new PlanetWars(message, 1);
			if (pw.Init() == 1) {
			    DoTurn(pw, r);
			}
			message = "";
		    } else {
			message += line + "\n";
		    }
		    line = "";
		    break;
		default:
		    line += (char)c;
		    break;
		}
	    }
	} catch (Exception e) {
	    // Owned.
	}
    }
}

File: planet_wars/starter_packages/python/MyBot.py
#!/usr/bin/env python
from planetwars import *

def do_turn(pw):
    my_strongest = -1
    their_weakest = 99999999
    for x in range (0, len(pw.planets)):
        planet = pw.planets[x]
        if planet.owner == 1 and planet.num_ships > my_strongest:
            my_strongest = x
        elif planet.owner != 1 and planet.num_ships < their_weakest:
            their_weakest = x

    if my_strongest != -1 and their_weakest != 99999999:
        pw.issue_order(my_strongest,
                       their_weakest,
                       pw.planets[my_strongest].num_ships / 2)
    pw.end_turn()

def main():
    map_data = ""
    while True:
        line = StringUtils.input()
        if line[0:2] == "go":
            do_turn(PlanetWars(map_data))
            map_data = ""
        else:
            map_data += line

if __name__ == "__main__":
    main()
